tests/typ/compiler/ops.typ

```diff
@@ -301,3 +301,41 @@
 #let inc2 = inc.with(y: 2)
 #test(inc2(2), 4)
 #test(inc2(2, y: 4), 6)
+
+---
+// Test chained comparison operations
+
+#test(1 < 2 < 3, true)
+#test(1 < 2 < 1, false)
+#test(1 < 2 <= 2, true)
+#test(1 < 2 <= 1, false)
+#test(1 <= 2 < 3, true)
+#test(1 <= 2 < 1, false)
+#test(1 <= 2 <= 2, true)
+#test(1 <= 2 <= 1, false)
+#test(1 != 2 == 2, true)
+#test(1 != 2 == 1, false)
+#test(1 == 2 != 3, false)
+#test(1 == 2 != 1, false)
+#test(1 == 2 == 2, false)
+#test(1 == 2 == 1, false)
+
+---
+// Test chained comparison with more complex expressions
+#let a = (1, 2, 3)
+#let b = (a: 1, b: 2, c: 3)
+
+#test(a.at(0) < a.at(1) < a.at(2), true)
+#test(a.at(0) < b.a * 2 < a.at(2), true)
+#test(a.at(0) < b.c * 2 < a.at(2), false)
+
+---
+// Test invalid chained operations
+
+// Error: 14-16 only transitive comparisons are allowed
+#test(1 != 2 != 3, true)
```

Comment:
I can't understand, why this is not allowed? Or, maybe there exists a more clear example:

```suggestion
#test(1 != 2 != 1, false)
```

```
1 != 2 -> true
2 != 1 -> true
but 1 != 1 is false
```

If I understand transitivity correctly

Reply:
The boolean value is kind of irrelevant as this is not allowed as per @MDLC01's comment